### ISSUE TITLE
Einrichtung eines Mailpostfaches funktioniert nicht

### ISSUE THREAD

**Issue by DerKaffeeschluerfer** [berliCRM/berlicrm#677]
Hallo zusammen,

ich habe das CRM soweit eingerichtet. Nun wollte ich im Mailmanager das Mailpostfach einrichten. Ich habe die Zugangsdaten und den Server etc eingetragen. Es dauert dann einige Zeig und dann kommt die Fehlermeldung, dass der Server nicht erreicht werden konnte, weil  angeblich die Zugangsdaten falsch sind. Die Zugangsdaten sind korrekt und ich habe alle erdenklichen Kombinationen der SSL/TLS Einstellungen probiert. Das Postfach funktioniert in einem Mailprogramm (Thunderbird) völlig problemlos. Benutzername und Passwort habe ich per Copy&Paste eingefügt. Tippfehler sind also ausgeschlossen. 

Ich bin über jeden Tipp dankbar. Eine Logdatei konnte ich leider nicht finden.

Viele Grüße
Stephen

**Comment by urban-thinking** [comment]
Hiya. Kannst du mal beispielhaft (also ohne zu viel zu verraten!) alle Werte hier posten und wie genau du sie eingetragen hast? Ist das ein "gängiger" oder ein sehr spezieller Mailserver? Irgendwas von Microsoft?

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Hallo Emilio,

der Mailserver ist imap.1blu.de (mein Webhoster). Ich hatte bisher nirgendwo Probleme mit dem Server. Ich verwende verschiedene Mailclients, Webmail etc und alles funktioniert bestens. Ich gehe ehrlich gesagt auch nicht davon aus, dass ich bei der Einrichtung einen Fehler gemacht habe ;-) Das sollte ein IT-Admin eigentlich gerade so hindekommen, denke ich :-D 

Gibt es keine Logdateien, bei denen man schauen kann, oder einen Debugmodus? Das würde vielleicht genauere Infos bringen. Ich hänge noch einen Screenshot hier dran...   

![grafik](https://user-images.githubusercontent.com/78486512/167432894-66a8452b-628b-4caf-9948-60394fdf43b7.png)

**Comment by urban-thinking** [comment]
Hmm ... wolltest du für das CRM einen SMTP-Sendeserver einrichten? Oder wolltest du externe Postfächer via IMAP auf bestimmte Inhalte scannen? Dein Screenshot zeigt die Config von letzterem. Den Sendeserver richtet man unter Zahnrad > andere Einstellungen > E-Mail Server ein.

Ein E-Mail Konverter (Mailscanner) Postfach habe ich eben auf einer frischen Installation testweise eingerichtet. Klappt.

![image](https://user-images.githubusercontent.com/11784828/167556807-45cfa1cb-62df-4a41-9277-dbc0cb2ffe50.png)

Kommst du auf die Shell von dem System, wo das CRM läuft? Gibt es evtl. ausgehend eine Firewall, die den IMAPS Port blockt?

Liebe Grüße ... Emilio

**Comment by urban-thinking** [comment]
Ahh ... sehe gerade, du schriebst "im Mailmanager" ... dann passt aber dein ScrenShot nicht dazu, der den E-Mail Konverter zeigt?! Ich habe in dem frischem System nun auch auch mal den Mailmanager probiert, klappt ebenfalls:

![image](https://user-images.githubusercontent.com/11784828/167557928-9dd79ede-fdc9-4146-a799-d460345b8e85.png)

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Hallo Emilio,

du hast Recht, es war im E-Mail Konverter. Ich wollte dort den IMAP Account einfügen. Im Mailmanager habe ich das noch nicht probiert. Einen SMTP-Account habe ich bereits eingerichtet.

Ich werde es gleich mal im Mailmanager probieren.Liebe Grüße
Stephen

**Comment by DerKaffeeschluerfer** [comment]
Ich hab es gerade schnell im Mailmanager probiert... Dort kommt die Meldung 

IMAP protocol error: Authentication aborted by client., IMAP Authentication cancelled 

Ich betreibe auf dem gleichen Server noch eine Nextcloud, die dort auch per IMAP Mails abruft. Selbstverständlich auch mit SSL Dort läuft IMAP über Port 993 und SMTP über 465 jeweils SSL/TLS.

Kann man die Ports irgendwo angeben?
LG
Stephen

**Comment by urban-thinking** [comment]
Im Eingabefeld **Servername** sollte so was tun:  mein.imapserver.de:993

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Wenn ich :993 im Feld Servername eingebe, kommt diese Fehlermeldung:

![grafik](https://user-images.githubusercontent.com/78486512/167606412-6dc4d000-d516-477a-8883-ea38d5c2b9dd.png)

**Comment by urban-thinking** [comment]
Vermutlich hast du TLS ausgewählt. Bei mir tut es so:

![image](https://user-images.githubusercontent.com/11784828/167608094-55d992d9-3760-44b7-bf34-eb8937536f2f.png)

Aber der Fehler zeigt ja auch, dass er bereits den gewünschten Port verwendet ;-)

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Ich habe alle Einstellungen probiert! Zertifikat prüfen ja/nein, SSL/TLS... In meiner Verzweifelung habe ich sogar IMAP2 probiert 😱

**Comment by urban-thinking** [comment]
In meiner Testumgebung taten alle genannten Varianten von mir. Ich hatte ja schon nach Shell-Zugriff und etwaiger ausgehender Firewall gefragt. Man sollte IMAPS vom betroffenen Server aus der Shell testen, z.B. mit mutt oder so ...

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Ja, ich hab Shell-Zugriff. Kann aber keine Software installieren... Wie gesagt, ich habe die Nextcloud mit E-Mail (IMAP und SMTP) da laufen... Und das gleiche Postfach läuft in der NC

**Comment by urban-thinking** [comment]
Also geht IMAPS bei dir weder im **Mailscanner** noch im **Mailmanager**? Das ist ja schräg. Hat das CRM die gleiche PHP-Version inkl. aller notwendigen Module wie auch deine NC? Überhaupt ... hast du die Installationsvoraussetzungen des CRMs bis in's kleinste Detail erfüllt? Das CRM ist da sehr "undankar", falls nicht ;-)

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Klar! Es ist der gleiche Server! die NC liegt im Ordner nc und das CRM im Ordner crm direkt nebeneinander... Ich kann micht nicht erinnern, dass bei der Installation etwas nicht passte. Es ist schon etwas her... Ich kann gerne nochmal eine frische Installation machen... Platz auf dem Server ist reichlich vorhanden

**Comment by urban-thinking** [comment]
Es geht weniger um eine "frische" Installation ... mehr darum, dass alles Voraussetzungen passen. Siehe auch https://blog.crm-now.de/doc/berliCRM/installation/Installation_berlicrm.html .

Frische Installation macht nur Sinn, falls irgendwas modifiziert wurde am Code oder so. Welche PHP-Version verwendest du?

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
PHP-Version... Ich sehe gerade, dass ich 7.4 nutze... Das könnte das Problem sein... Ich schau mal, ob ich auf 7.3 umstellen kann... Vor allem muss ich gucken, was die NC damit machen ;-)

**Comment by urban-thinking** [comment]
> PHP-Version... Ich sehe gerade, dass ich 7.4 nutze... Das könnte das Problem sein... Ich schau mal, ob ich auf 7.3 umstellen kann... Vor allem muss ich gucken, was die NC damit machen ;-)

Ja, das ist (noch) definitiv ein Problem und macht dann an "komischen Stellen" Mucken. Eine neue berliCRM Release steht gerade in den Startlöchern ... diese kann dann auch mit PHP 7.4 eingesetzt werden.

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
Also, auch mit PHP 7.3 klappt es nicht. Ich habe die anderen Voraussetzungen nochmal gesprüft. Das sollte soweit alles passen. Die max_execution_time war etwas niedriger... habe ich aber angepasst. gebracht hat es nichts ;-)

**Comment by DerKaffeeschluerfer** [comment]
Ich könnte dir Testweise mal ein Mailpostfach auf dem Server einrichten und gebe dir die Zugangsdaten... Dann könntest du es mal bei dir testen. Was mir gerade einfällt... Gibt es Einschrankungen bei der Passwortlänge? Ich verwende recht lange Passwörter...

**Comment by DerKaffeeschluerfer** [comment]
Also... Ich habe gerade ein Test-Postfach probiert. Hier habe ich nur ein Passwort mit 15 Zeichen verwendet. Bei dem anderen hatte ich 30... Bei Test-Postfach funktioniert es

**Comment by DerKaffeeschluerfer** [comment]
Also... Anscheinend gibt es beim Mailmanager und beim Mail Konverter die Einschränkung, dass ein Passwort maximal 15 Zeichen lang sein darf. Beim Mailversand gibt es diese Einschränkung nicht. Dort klappt es auch mit einem 30 Zeichen langen Passwort.

**Comment by Archibald111** [comment]
vermutlich reicht da der Speicherplatz von 255 Zeichen nicht aus: die Passwörter werden in der DB (table vtiger_mail_accounts) verschlüsselt abgelegt. Bitte mal mit einer Umstellung des Passwort Feldes in dieser table von varchar 255 auf MEDIUMTEXT versuchen.

**Comment by DerKaffeeschluerfer** [comment]
Hallo Frank,

ich kann das gerne ausprobieren. Aber dann muss ich mir ein Testsystem aufsetzen. Ich spiele nicht an der produktiven Datenbank rum... 

Ist denn das Feld für den SMTP Account in der Datenbank vom Typ mediumtext? An der Stelle hat das lange Passwort funktioniert. 

Am 10. Mai 2022 15:58:40 MESZ schrieb Frank Piepiorra ***@***.***>:

**Comment by Archibald111** [comment]
SMTP PW liegt in vtiger_systems mit VARCHAR 100, allerdings unverschlüsselt

**Comment by urban-thinking** [comment]
Habe das gerade mal getestet ... wenn man in der vtiger_mailscanner Table das Feld password von varchar(255) auf varchar(4000) setzt, dann klappt der Mailscanner auch mit den 30 Zeichen langen Passworten.

Liebe Grüße ... Emilio

**Comment by urban-thinking** [comment]
> SMTP PW liegt in vtiger_systems mit VARCHAR 100, allerdings unverschlüsselt

Jo ... das machte aber auch keine Probleme. Dass es dort unverschlüsselt liegt ist allerdings uncool.

Liebe Grüße ... Emilio

**Comment by DerKaffeeschluerfer** [comment]
> Jo ... das machte aber auch keine Probleme. Dass es dort unverschlüsselt liegt ist allerdings uncool.

Das wollte ich auch gerade schreiben...
